ada_feeding/ada_feeding/trees/acquire_food_tree.py

```diff
@@ -231,75 +325,206 @@ def create_tree(
-                            inputs={
-                                "position": BlackboardKey("extract_position"),
-                                "frame_id": "approach",
-                            },
-                            outputs={
-                                "constraints": BlackboardKey("goal_constraints"),
-                            },
-                        ),
-                        MoveIt2OrientationConstraint(
-                            name="ExtractOrientation",
-                            ns=name,
-                            inputs={
-                                "quat_xyzw": BlackboardKey("extract_orientation"),
-                                "frame_id": BlackboardKey("ee_frame_id"),
-                                "constraints": BlackboardKey("goal_constraints"),
-                            },
-                            outputs={
-                                "constraints": BlackboardKey("goal_constraints"),
-                            },
-                        ),
-                        MoveIt2Plan(
-                            name="ExtractPlan",
-                            ns=name,
-                            inputs={
-                                "goal_constraints": BlackboardKey("goal_constraints"),
-                                "max_velocity_scale": 0.8,
-                                "max_acceleration_scale": 0.8,
-                                "cartesian": True,
-                                "cartesian_max_step": 0.001,
-                                "cartesian_fraction_threshold": 0.95,
-                            },
-                            outputs={"trajectory": BlackboardKey("trajectory")},
-                        ),
-                        MoveIt2Execute(
-                            name="Extraction",
-                            ns=name,
-                            inputs={"trajectory": BlackboardKey("trajectory")},
-                            outputs={},
-                        ),
-                    ],
-                ),
-            ],
-        )
+                                py_trees.decorators.FailureIsSuccess(
+                                    name="GraspSuceed",
+                                    child=TimeoutFromBlackboard(
+                                        name="MoveIt2ServoTimeout",
+                                        ns=name,
+                                        duration_key=BlackboardKey("duration"),
+                                        child=py_trees.decorators.SuccessIsRunning(
+                                            name="GraspRetryInfinite",
+                                            child=py_trees.composites.Sequence(
+                                                name="GraspRetry",
+                                                memory=True,
+                                                children=[
+                                                    UpdateTimestamp(
```

Comment:
I think we use the `UpdateTimestamp` + `publishers.FromBlackboard` sequence enough (4x in this file alone) that it makes sense to make it an idiom.

Reply:
Done